src/sinks/elasticsearch/common.rs

```diff
@@ -77,6 +77,7 @@ impl ElasticsearchCommon {
                     .aws
                     .as_ref()
                     .map(|config| config.region())
+                    .ok_or(ParseError::RegionRequired)?
                     .ok_or(ParseError::RegionRequired)?;
```

Comment:
What's up with this?

Reply:
I figured this is just from whatever PR and it's fine, just looks off at a glance.

Reply:
Ah, yeah, this is a bit wonky but the first `ok_or` checks if `config.aws` is set. The second checks for `config.aws.region`. These could probably be reordered so that the first `ok_or` happens first to avoid the `map`. Something like: `config.aws.ok_or(ParseError::RegionRequired)?.region().ok_or(ParseError::RegionRequired)`.

I also opened #12534 to try to fix up and make consistent the AWS region handling.

Reply:
Makes sense - definitely hard to read without full context 😆